diff --git a/xbmc/platform/linux/PlatformLinux.h b/xbmc/platform/linux/PlatformLinux.h
index c45d41143a..9a872a5f55 100644
--- a/xbmc/platform/linux/PlatformLinux.h
+++ b/xbmc/platform/linux/PlatformLinux.h
@@ -21,7 +21,6 @@ public:
   ~CPlatformLinux() override = default;
 
   bool Init() override;
-  bool IsConfigureAddonsAtStartupEnabled() override { return true; };
 
 private:
   std::unique_ptr<OPTIONALS::CLircContainer, OPTIONALS::delete_CLircContainer> m_lirc;

