diff --git a/xbmc/platform/linux/PlatformLinux.h b/xbmc/platform/linux/PlatformLinux.h
index a7ea4ac..097b033 100644
--- a/xbmc/platform/linux/PlatformLinux.h
+++ b/xbmc/platform/linux/PlatformLinux.h
@@ -21,7 +21,6 @@ public:
   ~CPlatformLinux() override = default;
 
   bool InitStageOne() override;
-  bool IsConfigureAddonsAtStartupEnabled() override { return true; }
 
 private:
   std::unique_ptr<OPTIONALS::CLircContainer, OPTIONALS::delete_CLircContainer> m_lirc;
