diff --git a/configure.ac b/configure.ac
index d28c673..68d24af 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,7 +43,7 @@ AM_CONDITIONAL(HAVE_MAN2HTML,[
     man2html_cmd="man2html -r -M index.html doc/man-source/irexec.1"
     test $($man2html_cmd 2>/dev/null | wc -l) -gt 100])
 AS_IF([test x$HAVE_MAN2HTML_TRUE = x],AC_MSG_RESULT([yes]),AC_MSG_RESULT([no]))
-AC_CHECK_PROG([DOXYGEN],[doxygen],[yes],[no])
+AC_CHECK_PROG([DOXYGEN],[doxygen],[no],[no])
 AM_CONDITIONAL(HAVE_DOXYGEN, test x$DOXYGEN = xyes)
 LT_INIT([disable-static])
