diff --git a/tools/Linux/kodi.metainfo.xml.in b/tools/Linux/kodi.metainfo.xml.in
index 7b8a2037c0..eb70e2269c 100644
--- a/tools/Linux/kodi.metainfo.xml.in
+++ b/tools/Linux/kodi.metainfo.xml.in
@@ -37,7 +37,7 @@
         <keyword>tv</keyword>
         <keyword>television</keyword>
         <keyword>remote control</keyword>
-    </keyboards>
+    </keywords>
     <launchable type="desktop-id">@APP_NAME_LC@.desktop</launchable>
     <screenshots>
         <screenshot type="default">

